File: exercice6/main.ts
class Customer {

    name: string;
    accountNumber: string;

    constructor(name: string, accountNumber: string) {
        this.name = name;
        this.accountNumber = accountNumber;
    }
}

class CustomerPro extends Customer {
    siret: string;

    constructor(name: string, accountNumber: string, siret: string) {
        super(name, accountNumber);
        this.siret = siret;
    }
}

let switco: Customer = new Customer("Sweet Cookie", "14 539 202");
console.log(switco.name, switco.accountNumber);

let switCookie: CustomerPro = new CustomerPro("Sweet Cookie", "14 539 202", "802 606 582 00012");
console.log(switCookie.name, switCookie.accountNumber, switCookie.siret);

File: exercice6/main.java
class Playground {
    public static void main(String[ ] args) {
        Client switco = new Client("Sweet Cookie", 123445);
        ClientPro sweet = new ClientPro("Sweet Cookie", 545446, 802606582);
        System.out.println(switco.name + " " + switco.accountNumber);
        System.out.println(sweet.name + " " + sweet.accountNumber + " " + sweet.siret);
    }
}

class Client {
    String name;
    int accountNumber;
    
    Client(String name, int accountNumber) {
        this.name = name;
        this.accountNumber = accountNumber;
    }
}

class ClientPro extends Client {
    int siret;
    
    ClientPro(String name, int accountNumber, int siret) {
        super(name, accountNumber);
        this.siret = siret;
    } 
}